lib/liquid/standardfilters.rb

```diff
@@ -421,9 +421,9 @@ def at_most(input, n)
       result.is_a?(BigDecimal) ? result.to_f : result
     end
 
-    def default(input, default_value = '')
-      if !input || input.respond_to?(:empty?) && input.empty?
-        Usage.increment("default_filter_received_false_value") if input == false # See https://github.com/Shopify/liquid/issues/1127
+    def default(input, default_value = '', options = {})
+      options = {} unless options.is_a?(Hash)
+      if (options["allow_false"] == true ? input.nil? : !input) || (input.respond_to?(:empty?) && input.empty?)
```

Comment:
I think this would be easier to read:

```ruby
if options['allow_false']
  input.nil? ? default_value : input
else
  (!input || input.respond_to?(:empty?) && input.empty?) ? default_value : input
end
```

Reply:
I've modified this slightly to make it a little clearer. The equivalent in your example would need to be 
```
if options['allow_false']
  (input.nil? || input.respond_to?(:empty?) && input.empty?) ? default_value : input
else
  (!input || input.respond_to?(:empty?) && input.empty?) ? default_value : input
end
```